torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -221,15 +221,6 @@ def _input_node(gm: torch.fx.GraphModule, i: int) -> torch.fx.Node:
     raise IndexError(f"input {i} does not exist, only {seen} inputs in graph")
 
 
-def _can_detach(node: torch.fx.Node):
-    """
-    Avoid calling .detach() on inputs passed to _bind_nn_parameter()
-    """
-    from torch._dynamo.create_parameter_op import _bind_nn_parameter
-
-    return all(n.target is not _bind_nn_parameter for n in node.users)
```

Comment:
Won't you still hit this same issue?  How did you workaround it?

Reply:
I tried checking out a clean version of the code prior to my PR and deleting this function separately, and I didn't see anything fail when I ran `python test/inductor/test_distributed_patterns.py`. Is there another test that you expected to fail?

I was looking at an example a little more (running the `test_fake_distributed_inductor` test), and before my PR, the `_bind_nn_parameters` node looks like this:
```
_bind_nn_parameter = torch.ops.prims._bind_nn_parameter.default(cat, primals_2)
```

At first I thought the `can_detach()` logic here was trying to ensure that we don't detach `primals_2`, which corresponds to the sacrificial parameter (ATen graph [here](https://www.internalfb.com/phabricator/paste/view/P1203339086)) - but it looks like we are not detaching it, because we properly compute gradients for it in the backward graph [here](https://www.internalfb.com/phabricator/paste/view/P1203339864) (I think because we're tracing through this custom autograd.Function while we're tracing the joint graph, so we can tell that gradients should flow to sacrificial param).